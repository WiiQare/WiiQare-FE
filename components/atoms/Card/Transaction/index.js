import { CiCircleList, CiSaveDown2 } from "react-icons/ci";

import DropdownFilter from "../../../atoms/Dropdown/Filter";
import { Button } from "flowbite-react";
const CardTransaction = () => {
  return (
    <div className="border w-full rounded-lg py-8 px-6 flex flex-col flex-wrap md:flex-row justify-between gap-4 bg-white cursor-pointer">
      <div className="">
        <h1 className="md:text-2xl text-gray-700 font-bold flex-wrap flex items-center gap-2 text-lg">
          Transactions History
        </h1>
      </div>
      <div className="flex flex-row flex-wrap  gap-2 center justify-end space-x-2">
        <DropdownFilter
          label="Sort By"
          className="w-[auto]"
          labelClassName="py-1 w-[auto] border-0"
          icon={() => <span className="h-[15px] w-[15px] rounded bg-primary" />}
          items={["Detail", "Cancel"]}
        />
        <Button className="bg-primary">
          <CiSaveDown2 className="mr-2 h-5 w-5" />
          Download Report
        </Button>
        <DropdownFilter
          label="Sort By"
          className="w-[auto]"
          labelClassName="py-1 w-[auto]"
          icon={() => <CiCircleList />}
          items={["Detail", "Cancel"]}
        />
      </div>
    </div>
  );
};

export default CardTransaction;
