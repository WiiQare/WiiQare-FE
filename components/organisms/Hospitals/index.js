import React, { createContext, useState } from 'react';
import CardHeader from '../../atoms/Card/Header';
import { CiCircleList } from 'react-icons/ci';
import DataTable from 'react-data-table-component';
import { useSession } from 'next-auth/react';
import Fetcher from '../../../lib/Fetcher';
import { FcAddressBook, FcCellPhone } from 'react-icons/fc';
import Link from 'next/link';

const businessTypes = {
  MEDICAL_CABINET: { text: 'Cabinet Médical', color: '447698' },
  CLINIC: { text: 'Clinique', color: 'A143D3' },
  HOSPITAL: { text: 'Hôpital', color: 'FF5498' },
};

const columns = [
  {
    name: "IDENTITÉ DE L'HÔPITAL",
    selector: (row) => (
      <div className="flex py-4 gap-4 w-fit">
        <div className="min-w-20 w-20 h-fit border p-3 rounded-lg flex justify-center items-center">
          <img src={row.logoLink} alt="" />
        </div>

        <div className="flex flex-col gap-1">
          <span className="uppercase font-semibold text-md">{row.name}</span>

          <span className="text-xs text-gray-500">{row.email}</span>

          <span className="text-xs text-gray-500">{row.address}</span>

          <span className="flex gap-3 items-center text-sm text-gray-700 font-light mt-3">
            <span className="bg-orange opacity-90 px-2 py-1 text-white text-xs rounded-md w-fit flex gap-3 items-center">
              {row.packages ? row.packages.length : 0} Plan
              {row.packages.length > 1 ? 's' : ''}
            </span>
          </span>
        </div>
      </div>
    ),
  },

  {
    name: 'DÉTAILS BUSINESS',
    selector: (row) => (
      <div className="flex flex-col gap-4">
        <span
          className={`bg-gray-300 text-gray-900 py-1 pl-2 font-semibold w-fit pr-4 rounded-full opacity-80 `}
        >
          {businessTypes[row.businessType].text}
        </span>
        <div className="flex flex-col gap-1">
          <Link href={`tel:${row.phone}`} legacyBehavior>
            <span className="text-xs text-gray-500 flex items-center gap-1 cursor-pointer">
              <FcCellPhone size={18} /> {row.phone}
            </span>
          </Link>

          <Link href={`mailto:${row.email}`} legacyBehavior>
            <span className="text-xs text-gray-500 flex items-center gap-1 cursor-pointer">
              <FcAddressBook size={18} /> {row.email}
            </span>
          </Link>
        </div>
      </div>
    ),
  },

  {
    name: 'ACTIONS',
    button: true,
    cell: (row) => (
      <div>
        <button
          className="border border-gray-800 hover:bg-gray-800 hover:text-gray-200 font-semibold px-3 py-2 text-gray-800 text-xs rounded-md"
          onClick={(e) => alert(row.id)}
        >
          Packages
        </button>
      </div>
    ),
  },
];

const Hospital = () => {
  let { data: session } = useSession();
  const { data, isLoading, isError } = Fetcher(
    `/provider/list`,
    session.user.data.access_token,
  );

  return (
    <div className="p-2 space-y-6 md:py-8 md:px-6 mb-20 md:mb-6">
      <CardHeader
        title={'Liste des hôpitaux'}
        filter={{
          label: {
            title: 'Trier par',
            className: 'py-1 w-[auto]',
          },
          className: 'w-[auto]',
          icon: () => <CiCircleList />,
          items: ['Detail', 'Annuler'],
        }}
      />

      <div className="border rounded-lg overflow-x-auto w-full relative">
        {isLoading ? (
          <>Loading...</>
        ) : (
          <>
            {data.length > 0 ? (
              <DataTable
                columns={columns}
                data={data}
                fixedHeader
                className="px-8 bg-white"
                title="Liste des hôpitaux"
                persistTableHead
              />
            ) : (
              <div className="py-10 mx-auto w-1/2 flex flex-col justify-center items-center gap-4">
                <p className="text-gray-500 text-sm">
                  Aucun hôpital actuellement ajouté...
                </p>
              </div>
            )}
          </>
        )}
      </div>
    </div>
  );
};

export default Hospital;
