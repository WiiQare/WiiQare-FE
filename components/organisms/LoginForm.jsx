import React from "react";
import Link from "next/link";
import TextField from "@mui/material/TextField";
import Box from "@mui/material/Box";
import Stepper from "@mui/material/Stepper";
import Step from "@mui/material/Step";
import StepButton from "@mui/material/StepButton";
import Button from "@mui/material/Button";
import Stack from "@mui/material/Stack";
import useMediaQuery from "@mui/material/useMediaQuery";
import Visibility from "@mui/icons-material/Visibility";
import VisibilityOff from "@mui/icons-material/VisibilityOff";

import OnboardingScreen from "../molecules/OnboardingScreen";
import { useLocalStorage } from "../../hooks/useLocalStorage";
import {
  FormControl,
  IconButton,
  InputAdornment,
  InputLabel,
  OutlinedInput,
  Typography,
} from "@mui/material";
import MenuHolder from "../atoms/MenuHolder";
import BlinkSnackbar from "../atoms/BlinkSnackbar";

function LoginForm() {
  const matches = useMediaQuery("(max-width: 992px)");
  const [onboardingIsEnd, setOnboardingIsEnd] = React.useState("");
  const [showPassword, setShowPassword] = React.useState(false);

  const handleClickShowPassword = () => setShowPassword((show) => !show);

  React.useEffect(() => {
    setOnboardingIsEnd(localStorage.getItem("onboardingIsEnd") || "");
  }, []);

  const formHolder = (
    <>
      <div className="form-holder">
        <MenuHolder href="/register" label="SIGN UP" />
        <div className="signin-signup-form">
          <div className="form-items">
            <div className="form-title">Sign in to your account</div>
            <Box sx={{ mb: 3, mt: 2 }}></Box>
            <form id="signinform">
              <Stack spacing={2}>
                <TextField
                  id="outlined-basic"
                  fullWidth
                  label="E-mail Address"
                  variant="outlined"
                />
                <FormControl fullWidth variant="outlined">
                  <InputLabel htmlFor="outlined-basic1">Password</InputLabel>
                  <OutlinedInput
                    id="outlined-basic1"
                    label="Password"
                    type={showPassword ? "text" : "password"}
                    endAdornment={
                      <InputAdornment position="end">
                        <IconButton
                          aria-label="toggle password visibility"
                          onClick={handleClickShowPassword}
                          edge="end"
                        >
                          {showPassword ? <VisibilityOff /> : <Visibility />}
                        </IconButton>
                      </InputAdornment>
                    }
                  />
                </FormControl>
                <Box sx={{ textAlign: "right", mb: 2 }}>
                  <Typography color="text.secondary" variant="caption">
                    Forgot your password ?
                  </Typography>
                </Box>
                <Box>
                  <Button
                    size="large"
                    variant="contained"
                    onClick={() => {}}
                    className="bg-yellow"
                  >
                    SIGN IN
                  </Button>
                </Box>
              </Stack>
            </form>
          </div>
        </div>
      </div>
      <BlinkSnackbar />
    </>
  );
  return (
    <>
      {matches && !onboardingIsEnd ? (
        <OnboardingScreen
          onStartClick={() => {
            if (typeof window !== "undefined") {
              localStorage.setItem("onboardingIsEnd", "onboardingIsEnd");
            }
            setOnboardingIsEnd("onboardingIsEnd");
          }}
        />
      ) : (
        formHolder
      )}
    </>
  );
}

export default LoginForm;
