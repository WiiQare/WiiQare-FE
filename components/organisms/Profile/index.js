import Image from "next/image";
import CardHeader from "../../atoms/Card/Header";
import { BiCamera } from "react-icons/bi";
import Tabs from '@mui/material/Tabs';
import Tab from '@mui/material/Tab';
import Box from '@mui/material/Box';
import PropTypes from 'prop-types';
import Typography from '@mui/material/Typography';
import { useState } from "react";
import { FormControl, InputLabel, MenuItem, Select, TextField } from "@mui/material";

const TabHistories = [
	{
		name: "About"
	},

	{
		name: "Settings",
	}
]

const Profile = () => {
	const [value, setValue] = useState(0);

	const handleChange = (event, newValue) => {
		setValue(newValue);
	};
	return (
		<div className="p-2 space-y-6 md:py-8 md:px-6 mb-20">
			<CardHeader
				title={"My Profile"}
				breadcrumbs={[
					{
						item: "Home",
						link: "/"
					},
					{
						item: "John Doe",
						link: "/profile"
					}
				]}
				download={false}
				print={false}
			/>

			<section className="w-full flex flex-col gap-8 items-start pb-20 md:pb-0">
				<div className="w-full overflow-hidden md:col-span-2 rounded-lg p-4 flex flex-col gap-6 bg-white drop-shadow-sm">
					<div className="bg-[url(https://i.goopics.net/a5yedr.jpg)] bg-no-repeat h-56 rounded-lg"></div>

					<div className="flex justify-between items-center md:px-6">
						<div className="flex gap-5 items-center relative">
							<div className="w-20 h-20">
								<Image src={"https://xsgames.co/randomusers/avatar.php?g=male"} width={80} height={80} className="object-cover rounded-full h-full w-full" />
							</div>

							<div className="">
								<h1 className="text-sky font-bold text-xl">Mitchell C. Shay</h1>
								<span className="text-xs">frdrcpeter@gmail.com</span>
							</div>
						</div>

						<div>
							<button htmlFor="avatar" className="bg-[rgb(29,170,230,.6)] p-2 border-2 border-sky rounded-md text-white hover:bg-[rgb(29,170,230,.8)] w-fit">
								<BiCamera size={22} />
							</button>

							<input type="file" name="" id="avatar" className="hidden" />
						</div>
					</div>
				</div>

				<div className="md:w-3/4 w-full">
					<div className="min-full md:col-span-3 rounded-lg p-4 flex flex-col gap-4 bg-white drop-shadow-sm">
						<Box sx={{ width: '100%' }}>

							<Box sx={{ borderBottom: 1, borderColor: 'divider' }}>
								<Tabs value={value} onChange={handleChange} aria-label="basic tabs example">
									{TabHistories.map((item, index) => <Tab label={item.name} {...a11yProps(index)} key={index} />)}
								</Tabs>
							</Box>

							<div className="min-w-full">
								{TabHistories.map((item, index) => <TabPanelContent value={value} index={index} />)}
							</div>
						</Box>
					</div>
				</div>
			</section>
		</div>
	);
};

export default Profile;

export function TabPanel(props) {
	const { children, value, index, ...other } = props;

	return (
		<div
			role="tabpanel"
			hidden={value !== index}
			id={`simple-tabpanel-${index}`}
			aria-labelledby={`simple-tab-${index}`}
			{...other}
		>
			{value === index && (
				<Box sx={{ p: 3 }}>
					<Typography>{children}</Typography>
				</Box>
			)}
		</div>
	);
}

TabPanel.propTypes = {
	children: PropTypes.node,
	index: PropTypes.number.isRequired,
	value: PropTypes.number.isRequired,
};

function a11yProps(index) {
	return {
		id: `simple-tab-${index}`,
		'aria-controls': `simple-tabpanel-${index}`,
	};
}

function TabPanelContent({ value, index }) {
	return (
		<TabPanel value={value} index={index} >
			{
				index == 0 ? <About /> : <Settings />
			}
		</TabPanel>
	)
}

function About() {
	return (
		<section className="space-y-8">
			<h2 className="text-sky font-semibold">Personal Information</h2>

			<div className="md:w-3/6 space-y-4">
				<div className="grid grid-cols-2">
					<h5 className="font-semibold">Name :</h5>
					<span className="text-gray-500">Mitchell C. Shay</span>
				</div>

				<div className="grid grid-cols-2">
					<h5 className="font-semibold">Email :</h5>
					<span className="text-gray-500">frdrcpeter@gmail.com</span>
				</div>

				<div className="grid grid-cols-2">
					<h5 className="font-semibold">Availability :</h5>
					<span className="text-gray-500">Full time</span>
				</div>

				<div className="grid grid-cols-2">
					<h5 className="font-semibold">Age :</h5>
					<span className="text-gray-500">37</span>
				</div>

				<div className="grid grid-cols-2">
					<h5 className="font-semibold">Location :</h5>
					<span className="text-gray-500">Rosemont Avenue Melbourne, Florida</span>
				</div>

				<div className="grid grid-cols-2">
					<h5 className="font-semibold">Year Experience :</h5>
					<span className="text-gray-500">07 Year Experiences</span>
				</div>
			</div>
		</section>
	)
}

function Settings() {
	const [state, setState] = useState('');

	const handleState = (event) => {
		setState(event.target.value);
	};
	return (
		<section className="space-y-8">
			<h2 className="text-sky font-semibold">Account Setting</h2>

			<div className="flex md:grid md:grid-cols-2 gap-8">
				<TextField
					fullWidth
					type={"email"}
					className="placeholder:text-gray-400 hover:outline-none focus:ring-0 border border-gray-300 rounded-lg focus:ring-sky"
					label="Email Address"
					placeholder="Email Address"
					name="email"
					variant="outlined"
				/>

				<TextField
					fullWidth
					type={"password"}
					className="placeholder:text-gray-400 hover:outline-none focus:ring-0 border border-gray-300 rounded-lg focus:ring-sky"
					label="Password"
					name="password"
					placeholder="Password"
					variant="outlined"
				/>
			</div>

			<div className="">
				<TextField
					fullWidth
					type={"text"}
					className="placeholder:text-gray-400 hover:outline-none focus:ring-0 border border-gray-300 rounded-lg focus:ring-sky"
					label="Adress"
					placeholder="123 Main Street"
					name="email"
					variant="outlined"
				/>
			</div>

			<div className="">
				<TextField
					fullWidth
					type={"text"}
					className="placeholder:text-gray-400 hover:outline-none focus:ring-0 border border-gray-300 rounded-lg focus:ring-sky"
					label="Adress 2"
					placeholder="123 Main Street"
					name="email"
					variant="outlined"
				/>
			</div>

			<div className="flex md:hidden">
					<FormControl fullWidth>
						<InputLabel id="demo-simple-select-label">State</InputLabel>
						<Select
							labelId="demo-simple-select-label"
							id="demo-simple-select"
							value={state}
							label="State"
							onChange={handleState}
						>
							<MenuItem value={"Kinshasa"} selected={true}>Kinshasa</MenuItem>
							<MenuItem value={"Goma"}>Goma</MenuItem>
							<MenuItem value={"Lubumbashi"}>Lubumbashi</MenuItem>
						</Select>
					</FormControl>
				</div>


			<div className="flex md:grid md:grid-cols-4 gap-8">
				<TextField
					fullWidth
					type={"text"}
					className="col-span-3 md:col-span-2 placeholder:text-gray-400 hover:outline-none focus:ring-0 border border-gray-300 rounded-lg focus:ring-sky"
					label="City"
					placeholder="City"
					name="city"
					variant="outlined"
				/>

				<div className="hidden md:flex">
					<FormControl fullWidth>
						<InputLabel id="demo-simple-select-label">State</InputLabel>
						<Select
							labelId="demo-simple-select-label"
							id="demo-simple-select"
							value={state}
							label="State"
							onChange={handleState}
						>
							<MenuItem value={"Kinshasa"} selected={true}>Kinshasa</MenuItem>
							<MenuItem value={"Goma"}>Goma</MenuItem>
							<MenuItem value={"Lubumbashi"}>Lubumbashi</MenuItem>
						</Select>
					</FormControl>
				</div>

				<TextField
					fullWidth
					type={"text"}
					className="placeholder:text-gray-400 hover:outline-none focus:ring-0 border border-gray-300 rounded-lg focus:ring-sky"
					label="ZIP"
					name="zip"
					placeholder="ZIP"
					variant="outlined"
				/>
			</div>
		</section>
	)
}