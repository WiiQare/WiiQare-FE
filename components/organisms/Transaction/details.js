import Image from 'next/image';
import Link from 'next/link';
import { CiCircleInfo, CiCircleList } from 'react-icons/ci';
import { IoIosCall } from 'react-icons/io';
import ButtonBuy from '../../atoms/Button/Buy';
import CardHeader from '../../atoms/Card/Header';
import SimpleHeader from '../../atoms/Card/Header/simple';
import { Fragment, useEffect, useState } from 'react';
import { HiOutlinePlus } from 'react-icons/hi';
import { Dialog, Transition } from '@headlessui/react';

const DetailsTransaction = ({
  data,
  isLoading,
  isError,
  slug,
  fullname,
  email,
  phone,
}) => {
  const [count, setCount] = useState({ somme: 0, percent: 0, price: 0 });
  const [isOpen, setIsOpen] = useState(false);

  useEffect(() => {
    if (data && !data.code) {
      const somme =
        data &&
        data?.operations.reduce(
          (accumulateur, objet) => accumulateur + objet.amount,
          0,
        );
      const percent = (somme / data?.amount ?? 1) * 100;
      const month = data?.type == 'FEMME ENCEINTE' ? 9 : 12;
      console.log(month);
      const price =
        data?.amount /
        (data?.frequency == 'MONTH'
          ? month
          : data?.week == 'WEEK'
          ? month * 4
          : month * 31);

      setCount({ somme, percent, price });
    }
  }, [data]);

  const closeModal = () => {
    setIsOpen(false);
  };

  const openModal = () => {
    setIsOpen(true);
  };

  console.log(isLoading, isError, data);

  return (
    <div className="p-2 space-y-6 md:py-8 md:px-6">
      <CardHeader
        title={'Détails Épargne'}
        breadcrumbs={[
          {
            item: 'Épargne',
            link: '/saving',
          },
          {
            item: `#${slug.split('-')[0]}`.toUpperCase(),
            link: `/transactions/${slug}`,
          },
        ]}
        download={false}
        print={false}
      >
        <div className="flex flex-row-reverse gap-4">
          <Link href={`/saving/operation/${slug}`} legacyBehavior>
            <a
              className={`flex items-center gap-2 justify-between rounded-lg bg-orange py-3 text-white px-3 effect-up shadow h-fit`}
            >
              <span className="bg-white text-orange p-1 rounded-md">
                <HiOutlinePlus />
              </span>
              <span className="text-sm font-medium">Récharger le compte</span>
            </a>
          </Link>
        </div>
      </CardHeader>

      {isLoading || isError ? (
        <>Chargement...</>
      ) : (
        <section className="w-full grid md:grid-cols-3 gap-8 items-start pb-20 md:pb-0">
          <div className="w-full overflow-hidden md:col-span-2 rounded-lg py-8 flex flex-col gap-6 bg-white drop-shadow-sm">
            <div className="px-8 space-y-8">
              <div>
                <span className="text-gray-400 text-sm font-normal">
                  ID Épargne
                </span>
                <h2 className="font-semibold text-3xl">
                  #{slug.split('-')[0].toUpperCase()}
                </h2>
              </div>
              <div className="flex flex-col gap-4">
                <div className="overflow-x-auto">
                  <table className="table table-zebra text-xs w-full flex items-center">
                    <thead>
                      <tr>
                        <th>ID Paiement</th>
                        <th>Méthode de paiement</th>
                        <th>Montant</th>
                        <th>Type opération</th>
                        <th>Date</th>
                      </tr>
                    </thead>
                    {!data ||
                    !data?.operations ||
                    data?.operations?.length == 0 ? (
                      //<img src="https://i.goopics.net/z4ix58.png" alt="" className='opacity-60 w-20' />
                      <p className="text-center w-full py-2 text-gray-400 text-md">
                        Aucun versement...
                      </p>
                    ) : (
                      <tbody>
                        {data.operations.map((operation, index) => (
                          <tr key={index}>
                            <th>
                              #
                              {operation.id
                                .split('-')
                                [
                                  operation.id.split('-').length - 1
                                ].toUpperCase()}
                            </th>
                            <td>Carte bancaire</td>
                            <td>
                              {new Intl.NumberFormat('en-US', {
                                style: 'currency',
                                currency: operation.currency.toLowerCase(),
                              }).format(operation.amount)}
                            </td>
                            <td>{operation.type}</td>
                            <td className="capitalize">
                              {new Intl.DateTimeFormat('fr', {
                                dateStyle: 'medium',
                              }).format(new Date(operation.createdAt))}
                            </td>
                          </tr>
                        ))}
                      </tbody>
                    )}
                  </table>
                </div>
                <span className="w-full bg-gray-100 text-gray-500 h-fit p-2 rounded-lg flex gap-2 items-start text-sm">
                  <div>
                    <CiCircleInfo size={23} className="text-gray-400" />
                  </div>
                  <ul className="leading-6 font-light w-full pr-5 space-y-2">
                    <li className="flex justify-between w-full">
                      Objectif à atteindre:{' '}
                      <span className="text-primary font-bold">
                        {new Intl.NumberFormat('en-US', {
                          style: 'currency',
                          currency: data?.currency?.toLowerCase() ?? 'USD',
                        }).format(data.amount)}
                      </span>
                    </li>

                    <li className="flex justify-between w-full">
                      Cause d'épargne: <span className="">{data.type}</span>
                    </li>
                    <li className="flex justify-between w-full">
                      Frequence de paiement:{' '}
                      <span className="">
                        {new Intl.NumberFormat('en-US', {
                          style: 'currency',
                          currency: data?.currency?.toLowerCase() ?? 'USD',
                        }).format(count.price.toFixed(2))}{' '}
                        /
                        {data.frequency == 'MONTH'
                          ? 'mois'
                          : data.frequency == 'WEEK'
                          ? 'semaine'
                          : 'jour'}
                      </span>
                    </li>

                    <hr />

                    <li className="flex justify-between w-full">
                      Rester à atteindre:{' '}
                      <span className="font-bold">
                        {new Intl.NumberFormat('en-US', {
                          style: 'currency',
                          currency: data?.currency?.toLowerCase() ?? 'USD',
                        }).format(data.amount - count.somme)}
                      </span>
                    </li>
                  </ul>
                </span>
              </div>
            </div>

            <hr />

            <div className="px-8 space-y-6">
              <div>
                <h3 className="font-semibold text-xl">Recipients</h3>
              </div>

              <div className="flex flex-col md:flex-row justify-between gap-6 md:items-center">
                <div className="flex gap-3 items-center">
                  <div className="w-16 md:w-20">
                    <Image
                      src={`https://ui-avatars.com/api/?uppercase=true&background=random&name=${fullname}&bold=true&color=FFF`}
                      width={50}
                      height={50}
                      className="object-cover w-full h-full rounded-full"
                      alt="Avatar"
                    />
                  </div>
                  <div>
                    <h1 className="font-semibold text-lg md:text-2xl text-gray-900">
                      {fullname}
                    </h1>
                    <span className="text-xs md:text-sm text-gray-500">
                      {email}
                    </span>
                  </div>
                </div>

                <Link href={`tel:${phone}`} legacyBehavior>
                  <button className="flex gap-4 items-center border border-sky rounded-lg px-4 py-3 text-sky">
                    <IoIosCall size={40} />
                    <div className="flex flex-col gap-1 text-sm">
                      <span className="font-light">Phone Number</span>
                      <span className="font-bold">{phone}</span>
                    </div>
                  </button>
                </Link>
              </div>
            </div>
          </div>

          <div className="flex flex-col gap-6">
            <div className="w-full h-60 m-auto bg-red-100 rounded-xl relative text-white drop-shadow-sm ">
              <img
                className="relative object-cover w-full h-full rounded-xl"
                src="https://i.imgur.com/kGkSg1v.png"
                alt=""
              />

              <div className="w-full px-8 absolute top-6 flex flex-col gap-5">
                <div className="flex justify-between">
                  <div className="space-y-1">
                    <p className="font-extralight">{fullname}</p>
                    <h1 className="md:text-3xl font-bold">
                      {new Intl.NumberFormat('en-US', {
                        style: 'currency',
                        currency: data?.currency?.toLowerCase() ?? 'USD',
                      }).format(count.somme)}
                    </h1>
                  </div>
                  <div>
                    <Image
                      alt="logo"
                      className="w-20 h-16 object-contain"
                      src="/images/logo.png"
                      width={100}
                      height={100}
                    />
                  </div>
                </div>

                <div className="space-y-2">
                  <div className="pt-1">
                    <p className="font-medium tracking-more-wider text-center">
                      **** **** **** {slug.split('-')[1].toUpperCase()}
                    </p>
                  </div>
                  <div className="pt-5">
                    <div className="flex justify-around">
                      <div className="">
                        <p className="font-extralight text-xs">Date</p>
                        <p className="font-medium tracking-wider text-sm">
                          {new Intl.DateTimeFormat('fr', {
                            dateStyle: 'short',
                          }).format(
                            data?.createdAt
                              ? new Date(data?.createdAt)
                              : new Date(),
                          )}
                        </p>
                      </div>

                      <div className="">
                        <Link href={`/saving/operation/${slug}`} legacyBehavior>
                          <a
                            className={`flex items-center gap-2 justify-between rounded-lg bg-orange py-2 px-3 effect-up shadow`}
                          >
                            <span className="bg-white text-sky p-1 rounded-md">
                              <HiOutlinePlus />
                            </span>
                            <span className="text-sm font-light">
                              Récharger
                            </span>
                          </a>
                        </Link>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>

            <div className="rounded-lg py-4 px-4 flex flex-col gap-6 bg-white drop-shadow-sm">
              <SimpleHeader
                title={'Statistique'}
                describe={'Lorem ipsum dolor sit amet, consectetur'}
              />

              <div className="self-center">
                <div
                  className={`radial-progress ${
                    count.percent <= 50
                      ? 'text-purple'
                      : count.percent > 50 && count.percent <= 75
                      ? 'text-[#2ABB52]'
                      : 'text-[#441DE1]'
                  }`}
                  style={{ '--value': count.percent, '--size': '10rem' }}
                >
                  <span className="text-2xl">{count.percent.toFixed(0)}%</span>
                </div>
              </div>
            </div>
          </div>
        </section>
      )}

      <Transition appear show={isOpen} as={Fragment}>
        <Dialog as="div" className="relative z-50" onClose={closeModal}>
          <Transition.Child
            as={Fragment}
            enter="ease-out duration-300"
            enterFrom="opacity-0"
            enterTo="opacity-100"
            leave="ease-in duration-200"
            leaveFrom="opacity-100"
            leaveTo="opacity-0"
          >
            <div className="fixed inset-0 bg-black bg-opacity-25" />
          </Transition.Child>

          <div className="flex items-center justify-center min-h-full p-4 text-center">
            <div className="fixed inset-0 overflow-y-auto">
              {/* {state.type > 0 ? state.type == 2 ? <Toast type={"danger"} message={state.message} close={closeToast} /> : (state.type == 1 ? <Toast type={"success"} message={state.message} close={closeToast} /> : <></>) : <></>} */}

              <div className="flex items-center justify-center min-h-full p-4 text-center">
                <Transition.Child
                  as={Fragment}
                  enter="ease-out duration-300"
                  enterFrom="opacity-0 scale-95"
                  enterTo="opacity-100 scale-100"
                  leave="ease-in duration-200"
                  leaveFrom="opacity-100 scale-100"
                  leaveTo="opacity-0 scale-95"
                >
                  <Dialog.Panel className="w-full max-w-md p-6 overflow-hidden text-left align-middle transition-all transform bg-white shadow-xl rounded-xl space-y-10">
                    <div className="flex flex-col gap-6">
                      Lorem ipsum dolor sit, amet consectetur adipisicing elit.
                      Voluptatibus amet, ex at qui voluptate consequatur iste
                      reprehenderit voluptatum culpa quae vitae, laboriosam
                      minima placeat, odit ullam corrupti sunt suscipit quis?
                    </div>
                  </Dialog.Panel>
                </Transition.Child>
              </div>
            </div>
          </div>
        </Dialog>
      </Transition>
    </div>
  );
};

export default DetailsTransaction;
