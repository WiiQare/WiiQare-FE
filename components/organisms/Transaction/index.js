import React, { createContext, useState } from "react";
import CardHeader from "../../atoms/Card/Header";
import HistoryWallet from "../Wallet/History";
import { CiCircleList } from "react-icons/ci";

import Quickly from "./quickly";
export const TransactionContext = createContext();


const Transaction = () => {

	const [transaction, setTransaction] = useState({state: false, transactions: []});

	return (
		<div className="p-2 space-y-6 md:py-8 md:px-6 mb-20">
			<CardHeader
				title={"Historique des transactions"}
				sort={{
					icon: () => <span className="h-[15px] w-[15px] rounded bg-primary" />,
					items: ["DÃ©tail", "Annuler"]
				}}
				filter={{
					label: {
						title: "Trier par",
						className: "py-1 w-[auto]"
					},
					className: "w-[auto]",
					icon: () => <CiCircleList />,
					items: ["Detail", "Annuler"]
				}}
			/>

			<TransactionContext.Provider value={{ transaction, setTransaction }}>
				<HistoryWallet />
				<Quickly />
			</TransactionContext.Provider>

		</div>
	);
};

export default Transaction;
