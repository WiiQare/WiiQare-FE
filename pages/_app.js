import Head from "next/head";
import { SessionProvider } from "next-auth/react";
import { ThemeProvider } from "@mui/material/styles";
import { theme } from "../theme";

import "../styles/globals.css";
import "../styles/main.css";

export default function App({ Component, pageProps }) {
  const getLayout = Component.getLayout ?? ((page) => page);
  return (
    <>
      <Head>
        <meta
          name="description"
          content="Payé les soins à vos familles depuis l'étranger"
        />
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <link rel="icon" href="/favicon.ico" />
        <script src="https://unpkg.com/flowbite@1.5.3/dist/flowbite.js"></script>

      </Head>
      <SessionProvider session={pageProps.session}>
        <ThemeProvider theme={theme}>        
          {getLayout(<Component {...pageProps} />)}
        </ThemeProvider>
      </SessionProvider>
    </>
  );
}
