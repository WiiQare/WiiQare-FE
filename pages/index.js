import Head from 'next/head'
import { getSession, useSession, signOut } from "next-auth/react"

export default function Home() {

	const { data:session } = useSession()
	
	const handleSignOut = () => {
		signOut()
	}

	return (
		<>
			<Head>
				<title>JAMII - UNICEF CRYPTO</title>
				<meta name="description" content="Payé les soins à vos familles depuis l'étranger" />
				<meta name="viewport" content="width=device-width, initial-scale=1" />
				<link rel="icon" href="/favicon.ico" />
			</Head>
			

			{session ? <User session={session} handleSignOut={handleSignOut} /> : <Guest />}
		</>
	)
}

function User({session, handleSignOut}) {
	return (
		<main>
			Homepage
		</main>
	)
}

function Guest() {
	return (
		<main className='container mx-auto text-center py-20 space-y-8'>
			<h3 className="text-4xl font-bold">Guest User</h3>
		</main>
	)
}

export async function getServerSideProps({req}) {
	const session = await getSession({req})

	if(!session) {
		return {
			redirect: {
				destination: '/login',
				permanent: false
			}
		}
	}

	return {props: {session}}
}