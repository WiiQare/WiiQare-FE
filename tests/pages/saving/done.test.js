import Done from '@/pages/saving/done';
import { render } from '@testing-library/react';
import { TransactionContext } from '@/components/organisms/Transaction';
import { SessionProvider } from 'next-auth/react';
import { DrawContext } from '@/pages/_app';
import { QueryClientProvider, QueryClient } from 'react-query';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    query: {
      redirect_status: 'succeeded',
    },
  }),
}));

describe('Saving', () => {
  it('should render', () => {
    const queryClient = new QueryClient();
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <DrawContext.Provider value={{ draw: {}, setDraw: () => {} }}>
          <SessionProvider session={{ user: { data: { access_token: {} } } }}>
            <TransactionContext.Provider
              value={{ transaction: {}, setTransaction: () => {} }}
            >
              <Done />
            </TransactionContext.Provider>
          </SessionProvider>
        </DrawContext.Provider>
      </QueryClientProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
