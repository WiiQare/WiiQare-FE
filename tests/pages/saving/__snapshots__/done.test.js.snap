// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Saving should render 1`] = `
<div>
  <div
    class="flex flex-col gap-3 items-center mt-14"
  >
    <img
      alt="checkmark"
      class="w-36 opacity-90"
      data-nimg="1"
      decoding="async"
      height="80"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fi.goopics.net%2F22c5x7.png&w=256&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fi.goopics.net%2F22c5x7.png&w=96&q=75 1x, /_next/image?url=https%3A%2F%2Fi.goopics.net%2F22c5x7.png&w=256&q=75 2x"
      style="color: transparent;"
      width="80"
    />
    <span
      class="text-sm text-gray-600 "
    >
      Paiement effectué avec succès !
    </span>
  </div>
</div>
`;
